hw2/03-count.js

```diff
@@ -1,3 +1,18 @@
-// Add your code here
+// // Add your code here
+const box = document.getElementById('textBlock');
 
-input.addEventListener('keydown', handleKeyDown);
+const search = document.getElementById('search');
+const originalText = document.getElementById('textBlock').innerText;
+
+search.addEventListener('input', (event) => {
+  const searchText = event.target.value;
+  if (searchText === '') {
+    box.innerHTML = originalText;
+  } else {
+    const regex = new RegExp(searchText, 'gi');
+    let text = box.innerHTML;
+    text = text.replace(/(<mark>|<\/mark>)/gim, '');
+    const newText = text.replace(regex, '<mark>$&</mark>');
```

Comment:
While your code works, one small change that I can mention here is that I noticed that it highlights all the words that contain the input word, even if it's a part of a larger word. For example, when searching for '**in**', words like '**in**credulity' and 'spr**in**g' are also highlighted, which might not be the desired behavior as per the screenshots provided in HW2. Could you please take a look and make some modifications to ensure that only the exact input word is highlighted? 

I am also attaching the screenshot from our HW2 for reference.

<img width="688" alt="image" src="https://user-images.githubusercontent.com/114516950/234648823-9c14fc65-1c62-4b3a-bae7-0045f7ddd950.png">

Reply:
You're right, thanks for catching this. I will try to make this change